compiler/rustc_resolve/src/late.rs

```diff
@@ -1916,6 +1928,7 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
             };
             let missing_lifetime = MissingLifetime {
                 id: node_ids.start,
+                related_node_that_exists_in_source: finalize_node,
```

Comment:
Why not segment_id?

Reply:
No good reason other than I have overlooked it. I will push a fix soon, and probably rebase as well. It's been quite some time, plus #129510 will probably land soon.